### ISSUE TITLE
[📖 Docs]: Mocha third party UIs

### ISSUE THREAD

**Issue by ethanyou725** [webdriverio/webdriverio#12843]
### Pre-check

- [X] I know I can edit the docs but prefer to file this issue

### Describe the improvement

Unclear documentation

### Description of the improvement / report

Thanks for this awesome project. 
As I know, mocha could support [third party UIs ](https://github.com/mochajs/mocha/wiki/Third-party-UIs ), and webdriverIO could have mocha integration, but could we define third party UI with webdriverio mocha framework? I read the docs [https://webdriver.io/docs/frameworks/#using-mocha](https://webdriver.io/docs/frameworks/#using-mocha) and [https://github.com/webdriverio/webdriverio/tree/main/packages/wdio-mocha-framework#readme](https://github.com/webdriverio/webdriverio/tree/main/packages/wdio-mocha-framework#readme), but still very unclear about it.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christian-bromann** [comment]
@ethanyou725 can you elaborate on why you are interested to use a 3rd party UI for Mocha?

**Comment by ethanyou725** [comment]
> @ethanyou725 can you elaborate on why you are interested to use a 3rd party UI for Mocha?

Thanks for your kindly reply. Because we are migrating some exist test cases (*.spec.ts) to webdriverio.

e.g.
old case:
```
test('test case', async () => { await expect(1).toBe(1) })
```
 in mocha without custom ui, bdd style it must use `describe` and `it`,
and with custom ui interface, we could have the ability not change the spec file.

**Comment by christian-bromann** [comment]
Have you tried to set `mochaOpts.ui` to `tdd`?

**Comment by ethanyou725** [comment]
> Have you tried to set `mochaOpts.ui` to `tdd`?

yes,[ tdd ](https://mochajs.org/#tdd)provides different function suit, it could be more convenient if support custom ui.

**Comment by christian-bromann** [comment]
Actually with `qunit` you can run tests like this but it seems that reporters don't pick up the right test data for the reporting. Any contributions to improve this behavior would be appreciated.

**Comment by ethanyou725** [comment]
After dig into the ui interface implement in Mocha [here](https://github.com/mochajs/mocha/blob/master/lib/mocha.js#L369) and [there](https://github.com/mochajs/mocha/blob/master/lib/interfaces/bdd.js#L24),
I got a solution to use custom ui interface,
1. wdio.conf.js , set ui and require
```
mochaOpts: {
    ui: ['bddStep'],
    timeout: 60000 * 2,
    require: ['./src/mocha-step/step.cjs'],
  }
```
2. in the required module, we can implement the custom ui interface,  the interface name must be same.
```
module.exports = Mocha.interfaces.bddStep = function bddInterface(suite) {

}
```
And for reporting data, I will take a look later.
